control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -197,12 +231,22 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	for _, deletion := range updates.Consul.Deletions {
 		log.Info("deleting from Consul", "kind", deletion.Kind, "namespace", deletion.Namespace, "name", deletion.Name)
-		// TODO: the actual delete
+		r.cache.Delete(deletion)
 	}
 
 	for _, update := range updates.Consul.Updates {
 		log.Info("updating in Consul", "kind", update.GetKind(), "namespace", update.GetNamespace(), "name", update.GetName())
-		// TODO: the actual update
+		ref := translation.EntryToReference(update)
+		old := r.cache.Get(ref)
+		if cmp.Equal(old, update, buildOpts(update)) {
```

Comment:
I'm not super happy with cmp.Equal here, from reading the docs it can panic in weird ways, I'd prefer to write the comparison logic ourselves rather than rely on it